src/saturn/saturn_rom_extract.cpp

```diff
@@ -374,47 +378,101 @@ int saturn_rom_status(std::filesystem::path extract_dest, std::vector<std::strin
             else join_skyboxes(name_without_ext);
             continue;
         }
-        if (!std::filesystem::exists(extract_dest / entry.path)) {
+        if (!fs::exists(extract_dest / entry.path)) {
             needs_extract = true;
             if (todo != nullptr) todo->push_back(entry.path);
         }
     }
     if (type & EXTRACT_TYPE_SATURN) {
         for (const auto& entry : saturn_assets) {
-            if (!std::filesystem::exists(extract_dest / entry.path)) {
+            if (!fs::exists(extract_dest / entry.path)) {
                 needs_extract = true;
                 if (todo != nullptr) todo->push_back(entry.path);
             }
         }
     }
     if (!needs_extract) return ROM_OK;
-    while (true) {
-        rom_path = "";
+#ifdef FILE_PICKER
+    if (!fs::exists(gCLIOpts.RomPath)) {
+        open_file_picker();
+    }
+#else
         prompting_for_rom = true;
+#endif
+    while (true) {
+        rom_path = std::string(gCLIOpts.RomPath);
         while (prompting_for_rom) {
             std::this_thread::sleep_for(std::chrono::milliseconds(1));
         }
-        if (!std::filesystem::exists(rom_path)) {
+        if (!fs::exists(rom_path)) {
             pfd::message("Error", "A ROM at the specified path could not be found", pfd::choice::ok, pfd::icon::error).result();
+            prompting_for_rom = true;
             continue;
         }
-        uint32_t size = std::filesystem::file_size(rom_path);
+        uint32_t size = fs::file_size(rom_path);
         std::ifstream stream = std::ifstream(rom_path, std::ios::binary);
         unsigned char* data = (unsigned char*)malloc(size);
         stream.read((char*)data, size);
         stream.close();
         unsigned int checksum = crc32(data, size);
         if (checksum != ROM_CHECKSUM) {
             pfd::message("Error", "The specified ROM is invalid.\nMake sure it's not corrupted, extended or from the wrong region.", pfd::choice::ok, pfd::icon::error).result();
+            prompting_for_rom = true;
             continue;
         }
         break;
     }
     return ROM_NEED_EXTRACT;
 }
 
+// AppImage AppDir
+// copy external files from out of read-only AppImage to read-write user folder, 
+// usually ~/.local/share/v64saturn
+int copy_custom_assets(void) {
+    // AppImage environment variable
+    const char *appdir_cstr = std::getenv("APPDIR");
+    const char *exe_path_cstr = sys_exe_path();
+    const char *user_path_cstr = sys_user_path();
+    std::string srcdir, destdir = std::string(user_path_cstr);
+
+    // This affects non-AppImage builds also which means that if I leave it this way,
+    // all AppImage support might need to be toggled by a condition.
+    if (appdir_cstr == NULL) {
+        srcdir = std::string(exe_path_cstr);
+    }
+    else {
+        srcdir = std::string(appdir_cstr) + "/custom_assets/";
+    }
```

Comment:
I think for this version to work fully on Windows, probably there has to be a way to 1`00% reliably detect a read-only folder to copy the assets from and I don't know how to do that on Windows right now. This design tends to work in UNIX-like packages but I guess Windows is different

Reply:
When I look at it I'm not sure why exactly this produced "Failed to copy the required custom asset files for Saturn!" on Windows, when it doesn't on Linux even outside of AppImage, so more designing and testing for Windows is probably needed